### ISSUE TITLE
Grading system breaks down if there isn't any graded items filling every category

### ISSUE THREAD

**Issue by shusain93** [shusain93/GradeA#2]
They grading system should be moved to a different class file. 

The issue is when a setup for a class is like

Participation 25%
--Day 1: 100%
--Day 2: 95%

Other work 75%
--*dust*

should give a grade of 97.5%, instead it gives 48.75% which is wrong.

Solution would be to first check if all of the categories are filled, if not, look at it like it's 100%? Shitty solution, but I don't know how else to fix it.

**Comment by shusain93** [comment]
So I've got to possible solutions, neither which are particularly acceptable. Switch to a totally point based system which just doesn't make sense for some schools or do math logic to drop off categories and distribute their weight evenly against all other filled categories which would require a brand new grading class and a crap ton of logic + math. Actually, scratch that, three options. I can do the bad and broken way of just defaulting 100%s to every category that doesn't get filled which I would still need the new grading class for. So 2/3 of the options require a new class. Looks like this is going to be a long night.

**Comment by shusain93** [comment]
Alright so I've started to implement the category dropping method. I'm going to bed right now since it's 11:13 pm and I'm going to need to get early, but I've made it past the second most difficult part. I've gotten through assigning assignments into the categories based on their grading category ID which I'm glad I implemented. No idea when, but super glad I did.

My next steps for next time is to move the logic right now in the main method out into separate methods for ease of fixing and so I can read it. Next, we need to iterate and check if any of the categories are empty and go from there. Follow the notability diagram.
